### PR TITLE
Add new line before 'where' constraints in Quick Info

### LINKED ISSUE #5: Intellisense Tooltip get too long in case of many generic parameters
Hi, I hope to be in the right place, if not feel free to close this.

I noticed that Intellisense tool-tips become too long in case of many generic parameters with constraint, see image below.
intellisense tooltip

What I would like to see, if there are more than 1 generic constraints is to split every constraint in a new line (as you can see in generic constraint declaration on the class itself) to avoid creating a very long tool-tip and to gain readability

M.

### PR DESCRIPTION
**Pull request by Youssef1313**
Closes #5 (the oldest open issue :smile:)

- Tests to be added.
- VB to be fixed as well.

Open question:

- I could have done this in SymbolDisplay in the compiler layer. But I'm not sure if this would be okay.
- If this is to be done in the compiler layer, it can be done in two ways:

    - Always add line break.
    - Introduce a new SymbolDisplayOptions option, for example `SymbolDisplayMiscellaneousOptions.LineBreakBeforeTypeParameterConstraint` (will be a new public API).

The current approach is to do the work in IDE only.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Pics please. For simple and complex cases.

**Comment by Youssef1313** [comment]
Testing revealed that this doesn't handle named types. Named types take a different code path (`AddSymbolDescription(INamedTypeSymbol symbol)`). Here is how it looks like for methods:

image

I'll adjust the implementation and add unit tests once an approach is decided:

- Compiler layer - similar to https://github.com/dotnet/roslyn/pull/56025
- IDE layer - similar to the current state of PR.

**Comment by DoctorKrolic** [comment]
@Youssef1313 Looking at your screenshot, I would say, that you should add some offset (`\t` for instance) before each `where` line to make it even more readable

**Comment by Youssef1313** [comment]
image

@CyrusNajmabadi Let me know what do you think

**Comment by sharwell** [comment]
Can you also add a test where someone has a comment like this?

```csharp
/// <summary>
/// This is some text <see langword="where"/> I shouldn't have a line break.
/// </summary>
```

**Comment by DoctorKrolic** [comment]
@Youssef1313 Can you please revisit this PR. When a class has _really_ many `where`s it gets as ridiculous as this:
Npn9h4xeCf
I don't know how about you, but I can barely read this mess...

**Comment by Youssef1313** [comment]
@DoctorKrolic I'm getting a little bit busy as my university exams are approaching. :(
I don't mind if you want to take this up and open a PR (feel free to cherry-pick the commit or copy the code here and continue on it, if you want)

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 do you want to continue this pr?

**Comment by CyrusNajmabadi** [comment]
I'll take over this PR.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs [hunk 1]
```diff
@@ -478,28 +501,6 @@ private void AddDescriptionForNamedType(INamedTypeSymbol symbol)
             }
         }
 
-        private void AddSymbolDescription(INamedTypeSymbol symbol)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs:481]: inlined this.

**Comment by CyrusNajmabadi** [comment]
Not a problem at all :)

**Pull request merged**